string.go

```diff
@@ -166,3 +173,11 @@ func removeControlRune(r rune) rune {
 
 	return r
 }
+
+func removeNullChars(r rune) rune {
+	if r == nullRune {
+		return -1
+	}
+
+	return r
+}
```

Comment:
Probably worth adding a short doc here to explain the magic

```suggestion
// this function can be provided to strings.Map to remove null characters from a string
// When it returns a negative value the character is dropped from the string with no replacement.
// see: https://pkg.go.dev/strings#Map
func removeNullChars(r rune) rune {
	if r == nullRune {
		return -1
	}

	return r
}
```